browser/src/control/jsdialog/Util.Dropdown.js

```diff
@@ -43,10 +43,12 @@ JSDialog.OpenDropdown = function (id, popupParent, entries, innerCallback, popup
 
 	if (popupParent && !popupParent._onClose) {
 		popupParent._onClose = () => {
-			popupParent.setAttribute('aria-expanded', false);
+			popupParent.firstChild.setAttribute('aria-expanded', false);
```

Comment:
New code tries to guess the internal structure of "popupParent".
What if it will not have any children?

Can we do it with similar pattern as disabled state?
In JSDialog we have pattern that we setup property on the root container of any widget (just popupParent here as it was). But the widget itself is responsible to apply other changes in it's internal structure.

See: https://github.com/CollaboraOnline/online/blob/ae8300e5f556ce914e2c9256b6b2276a53520678/browser/src/control/Control.JSDialogBuilder.js#L1368
you could have similar Util file for a11y:
https://github.com/CollaboraOnline/online/blob/ae8300e5f556ce914e2c9256b6b2276a53520678/browser/src/control/jsdialog/Util.StateChange.ts#L35

Reply:
I haven’t seen additional widgets that combine a dropdown with an input and a button. If this widget is unique, it should work; however, ideally, each widget should function independently. I’ll review the synchronizeDisabledState code to determine if I can apply changes to maintain independence. Otherwise, code refactoring is beyond the scope of my tasks.